modules/BiomesAPI/src/main/java/org/terasology/biomesAPI/BiomeManager.java

```diff
@@ -89,14 +100,21 @@ public void setBiome(Biome biome, Vector3i pos) {
 
     @Override
     public void setBiome(Biome biome, CoreChunk chunk, Vector3i pos) {
-        setBiome(biome, chunk.chunkToWorldPosition(pos));
+        Vector3i position = chunk.chunkToWorldPosition(pos);
```

Comment:
This is exactly what should not be happening. Methods taking chunk as a parameter should rely on chunk-relative coordinates.